### PR TITLE
chore: change banner at top of v3 (CODAP-1319)

### PR DESCRIPTION
**Pull request by emcelroy**
[CODAP-1319](https://concord-consortium.atlassian.net/browse/CODAP-1319)

Replaces the Beta banner with a more flexible Announcement banner. The announcement banner references a JSON file stored in AWS S3 for content, allowing for modification without a new release of CODAP.

The new banner code borrows heavily from the [CFM v1](https://github.com/concord-consortium/cloud-file-manager/tree/v1.9.x) banner. Commit [50bf601](https://github.com/concord-consortium/codap/pull/2579/commits/50bf601d4bc54663c50ece09d00882e6c86f1011) specifically drops several style config options the CFM provides. Those don't seem necessary here. They can be added back if others disagree, though.

[CODAP-1319]: https://concord-consortium.atlassian.net/browse/CODAP-1319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: v3/src/components/announcement-banner/announcement-banner-utils.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ * Validate banner JSON has required fields.
+ */
+export function isValidBannerConfig(data: unknown): data is BannerConfig {
+  if (typeof data !== "object" || data === null) return false
+  const obj = data as Record<string, unknown>
+  return typeof obj.message === "string" &&
+         obj.message.length > 0 &&
+         typeof obj.id === "string" &&
+         obj.id.length > 0
+}
```
[the remark this answers is not in the document]
Comment (emcelroy) [reply v3/src/components/announcement-banner/announcement-banner-utils.ts:127]: Addressed in [30d6bbd](https://github.com/concord-consortium/codap/pull/2579/commits/30d6bbd38a8626e0e61a670478d41190974f69fe).

**Review by kswenson — APPROVED** [review]
👍 Looks good -- the following review was developed in conjunction with Claude Code 🤖.

## PR Review Summary — #2579 chore: change banner at top of v3 (CODAP-1319)

**Changes:** 19 files, +1126 / -154 lines

### What it does

Replaces the static beta banner with a flexible "Announcement" banner. The new
banner pulls its content from a JSON file in S3 (`/codap-resources/notifications/
v3-announcement-banner.json`), so the message can be changed, enabled/disabled, or
date-scoped without cutting a new CODAP release. The implementation borrows from
the CFM v1 banner, dropping a few style-config options.

### The approach

- **`announcement-banner-utils.ts`** — pure logic: fetch + validate config, parse
  markdown-style links, date-range gating, localStorage dismissal.
- **`announcement-banner.tsx`** — the component: fetches on mount, renders message
  with optional links + Feedback button + two dismiss controls (X = session-only,
  "Don't show again" = persisted to localStorage).
- **`ui-state.ts`** — `shouldRenderBetaBanner` → `shouldRenderAnnouncementBanner`,
  adding a `!hasInteractiveApiContext()` gate so it never shows in LARA/embedded.
- **`app.scss`** — layout-shift handled by a `has-announcement-banner` class
  toggled on `<html>` via `useLayoutEffect`, replacing the old `.beta` class.
- Beta banner/button files deleted; Cypress updated (see Assessment).

The `isBeta()` gate is intentionally dropped — the banner now shows in all builds
and visibility is fully controlled by the remote config's `enabled`/date fields.

### Assessment

This is a clean, well-engineered PR. Strengths:

- **Security:** message text rendered through React (escaped); links restricted to
  `https://` with `rel="noopener noreferrer"`; `buttonUrl` validated; non-https
  links degrade to plain text.
- **Fail-safe / fail-closed:** missing/404 JSON → no banner; strict runtime type
  checks on every optional field reject malformed config rather than rendering it
  anyway (commit 30d6bbd specifically hardens this).
- **Accessibility:** `role="status"`, `aria-label`s, visible focus outlines.
- **Test coverage:** excellent — 378-line utils suite, 252-line component suite,
  plus 2 Cypress integration tests for real-browser layout/lifecycle.
- **CI:** Build, Lint, Jest, full regression (368 tests) all green. One Cypress
  flaky-test marker, unrelated.

### Issues

No blocking issues. Minor observations only:

1. **Layout shift (CLS).** The banner appears asynchronously after the S3 fetch
   resolves, at which point the whole app shifts down 50px. The old beta banner
   was synchronous (`isBeta()` known at load), so there was no shift. This is an
   inherent tradeoff of remote config — likely acceptable, but worth a conscious
   nod. (Not fixable without blocking first paint on the fetch.)

2. **`parseMessageWithLinks` regex** uses `[^)]+` for the URL, so a link URL that
   itself contains `)` would be truncated. Edge case; we control the config, so
   low risk.

3. **`cy.wait(500)`** hard wait in the second Cypress test. Acceptable here since
   it's asserting *absence*, but it's the usual flake-prone pattern.

4. **Deploy dependency:** the banner JSON path is `notifications/v3-announcement-
   banner.json` under codap-resources. Safe to merge before the file exists
   (404 → no banner), but confirm the path matches the intended S3 location.
